frontend/src/lib/integrations/HubSpotIntegrationHelpers.tsx

```diff
@@ -0,0 +1,40 @@
+import api from 'lib/api'
+import { LemonBanner } from 'lib/lemon-ui/LemonBanner'
+import { Link } from 'lib/lemon-ui/Link'
+
+import { HogFunctionInputSchemaType, IntegrationType } from '~/types'
+
+export function HubSpotScopesWarning({
+    integration,
+    schema,
+}: {
+    integration: IntegrationType
+    schema?: HogFunctionInputSchemaType
+}): JSX.Element {
+    const requiredScopes = schema?.requiredScopes?.split(' ') || []
+    const missingScopes = requiredScopes.filter((scope: string) => !integration.config.scopes?.includes(scope))
+
+    if (integration.kind !== 'hubspot' || missingScopes.length === 0 || integration.config.scopes === undefined) {
+        return <></>
+    }
+    return (
+        <div className="p-2">
+            <LemonBanner
+                type="error"
+                action={{
+                    children: 'Reconnect',
+                    disableClientSideRouting: true,
+                    to: api.integrations.authorizeUrl({
+                        kind: integration.kind,
+                        next: window.location.pathname,
+                    }),
+                }}
+            >
+                Required scopes are missing: [{missingScopes.join(', ')}]. Note that some features may not be available
```

Comment:
Yes, it could work for multiple oauth integrations, but the way to get the authorized scopes is really different as it seems:

1. Slack - In the `x-oauth-scopes` headers from the token info URL.
2. Hubspot - Under the `scopes` key from the token info URL.
3. Google - Under the `scope` key from the token info URL OR under the `scope` key from the token URL.

Reply:
I've pushed some changes to generalize the logic. We'll grab authorized tokens from multiple possible locations:
[`370d97a` (#26738)](https://github.com/PostHog/posthog/pull/26738/commits/370d97add8474739812c5dd0bd3df22ef4b0ac02#diff-1641c1ea840c6fc781319690ce15e0910317cb5666615dd1f0e6ecef99227e5bR14-R30)